Fix notifications

sh-zam opened the pull request:
This _should_ most make the notifications work like it worked in past. (However, it doesn't use the pynotify).

There is one issue, the path to the icon is hardcoded. I am not sure, if we can dynamically access the installed location of hamster or the "future" installation structure of hamster.

CC: issue#493

ederag commented:
Thanks, this looks very interesting.
But for reasons explained in #493, this should be external (not in hamster itself):
> All these features can be implemented separately,
talking to the main hamster through dbus, for instance.

Please do ask there for any clarifications.

The next weeks free time will be devoted to finalizing v3.0, 
only then will I be able to help setting up an external service.

Meanwhile, please do keep this PR open.
If you merge master into it regularly, people will be able to use it ([instructions](https://github.com/projecthamster/hamster/wiki/How-to-contribute#one-shot-test)),
until the external companion is up and running.

About the icon location, the base path can be accessed in `defs`:
```python
>>> from hamster import defs
>>> defs.DATA_DIR
'/usr/share'
```
But that's about it.  The new icon installation path is set in 
https://github.com/projecthamster/hamster/blob/a572ddcaab1a094ee3bcfa3f8e14ad66ec691582/data/wscript_build#L7-L14

The directory and names changed in #485, 
so beware that it seems to work on your system because of remnants from a previous installation.

There should be a better way though, for instance in the ui, icons can be found by name:
https://github.com/projecthamster/hamster/blob/a572ddcaab1a094ee3bcfa3f8e14ad66ec691582/data/edit_activity.ui#L18

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/projecthamster/hamster?pullRequest=521) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/projecthamster/hamster?pullRequest=521) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/projecthamster/hamster?pullRequest=521) it.</sub>

ederag commented:
There will never be any CLA on this code. Please disregard the wrong checks (#589).

ederag commented:
There will never be any CLA on this code. Please disregard the wrong checks (#589).

sh-zam commented:
I have updated this to the current master and I have updated the way to get icon path, however I am not sure if this is how you're supposed to do it.

mwilck commented on src/hamster-cli.py:
```diff
@@ -46,7 +46,7 @@
 from hamster.lib import datetime as dt
 from hamster.lib.fact import Fact
-
+from hamster.lib.notifsmanager import notifs_mgr
```
`notifs_mgr` seems to be referenced nowhere in this file. What am I missing?

sh-zam replied:
Must be remnants of my debugging session. I'll remove it, thanks!

sh-zam commented:
Thank you for reviewing!

> I'd like to understand how it relates to the previous code removed in [08eaf77](https://github.com/projecthamster/hamster/commit/08eaf77eb925a1bd0c23bad0bcdb4db7310bfa5c) (`desktop.py`). Have you used that code as a template / reference somehow? 

No, I did not I'm afraid. I wrote it from scratch by looking at the docs :(

> You seem to have implemented some things differently (looking at the actual `Notify()` code for example - missing `transient` and `urgency` properties), could you elaborate a bit on that, either in the commit message or in comments in the code?

I didn't know that such a thing (`transient`) existed. I will update my patch to include it and add a few comments along the way.

PS: Just yesterday, I added code to close notifications automatically. It was an ugly hack. Thanks for mentioning `transient` :-)
